### DISCUSSION TITLE
ESP-IDF under C++ using nlohmann/json library

### DISCUSSION THREAD

**Discussion by Teodor12** [nlohmann/json#4359] [General]
Good afternoon!

Is there anybody who used/uses this library with esp-idf? I've just started a new hobby project for my ESP-32 board in C++. One of my friends recommended this library for handling json in C++. Can someone help me in "correctly importing" this library to my C++ project(I added as a git submodule, and getting horribly long linker errors)? Did I miss something?

Thank you for your answers!

Have a nice day!

**Comment by nlohmann** [comment]
The library is header-only, so it should work fine as an ESP-IDF component in principle — long linker errors usually mean the header isn't visible to all translation units that need it, or a component.mk/CMakeLists.txt include-path issue rather than anything JSON-specific. Could you share the exact linker error and how you registered the component (`idf_component_register` include dirs)? Also see discussion #4250 where another user shares an ESP-IDF/PlatformIO component setup that might help.

---
*This reply was drafted by [Claude Code](https://claude.com/claude-code) on behalf of @nlohmann, as part of a review of open discussions.*
